table/tables/tables.go

```diff
@@ -664,6 +665,9 @@ func (t *TableCommon) AddRecord(sctx sessionctx.Context, r []types.Datum, opts .
 		if err != nil {
 			return nil, err
 		}
+	} else if hasExplicitRowID {
+		// if _tidb_rowid is explicitly assigned, we need to rebase the allocator
+		err = t.RebaseAutoID(sctx, recordID.IntValue(), true, autoid.RowIDAllocType)
```

Comment:
should take `shard rowid` into consideration.. Ignore the leading bits of `shard rowid`